pypesto/sample/sample.py

```diff
@@ -67,14 +75,41 @@ def sample(
     if n_samples is not None:
         n_samples = bound_n_samples_from_env(n_samples)
 
+    if warm_start > 1.0 or warm_start < 0.0:
+        raise ValueError("warm_start must be in [0, 1].")
+
     # try to find initial parameters
     if x0 is None:
-        result.optimize_result.sort()
-        if len(result.optimize_result.list) > 0:
-            x0 = problem.get_reduced_vector(
-                result.optimize_result.list[0]["x"]
-            )
-        # TODO multiple x0 for PT, #269
+        if warm_start == 0:
+            logger.info("Sampling initial points from prior.")
+            get_start_params = PriorStartpoints(check_fval=True)
+            x0 = get_start_params.sample(
+                n_starts=1,
+                lb=problem.lb,
+                ub=problem.ub,
+                priors=problem.x_priors,
+            )[0]
+        elif result.optimize_result is not None:
+            result.optimize_result.sort()
+            if len(result.optimize_result.list) > 0:
+                x0 = problem.get_reduced_vector(
+                    result.optimize_result.list[0]["x"]
+                )
+            if warm_start < 1.0 and x0 is not None:
+                logger.info(
+                    f"Initializing sampling with a warm start from optimization "
+                    f"and prior sampling with weight: {warm_start}."
+                )
+                get_start_params = PriorStartpoints(check_fval=True)
+                x0_prior = get_start_params.sample(
+                    n_starts=1,
+                    lb=problem.lb,
+                    ub=problem.ub,
+                    priors=problem.x_priors,
+                )[0]
+                x0 = warm_start * x0 + (1 - warm_start) * x0_prior
+        else:
+            logger.info("No initial point provided.")
```

Comment:
Should be an exception with message e.g. "set warm_start to 0"? Or how can the "warm start" setting be satisfied?